src/components/avail-chart.component.ts

```diff
@@ -0,0 +1,389 @@
+import { Component, OnInit, OnDestroy, OnChanges, SimpleChanges, Input, Output, EventEmitter, ViewChild } from '@angular/core';
+import { Http, RequestOptions, Headers, Response } from '@angular/http';
+
+import { Observable } from 'rxjs/Observable';
+import { Subscription } from 'rxjs/Subscription';
+import { IntervalObservable } from 'rxjs/observable/IntervalObservable';
+
+import { IAvailDataPoint, TransformedAvailDataPoint, UrlType, TimeRange, FixedTimeRange, getFixedTimeRange,
+  isFixedTimeRange, Range, TimeInMillis } from '../model/types'
+import { ChartLayout } from '../model/chart-layout'
+import { ComputedChartAxis } from '../model/computed-chart-axis'
+import { xAxisTimeFormats } from '../util/utility'
+import { initTip } from '../util/avail-tip'
+
+declare const d3: any;
+declare const moment: any;
+declare const console: any;
+
+// TODO: remove? (was: with room for label)
+const MARGIN = { top: 20, right: 0, bottom: 10, left: 45 };
+
+@Component({
+  selector: 'hk-availability-chart',
+  template: `<div #target class='hawkular-charts'></div>`
+})
+export class AvailChartComponent implements OnInit, OnDestroy, OnChanges {
+
+  @ViewChild('target') target: any;
+
+  @Input() metricUrl: UrlType;
+  @Input() metricId = '';
+  @Input() metricTenantId = '';
+  @Input() authHeader: string;
+  @Input() refreshIntervalInSeconds = 5;
+  @Input() data?: IAvailDataPoint[];
+  @Output() timeRangeChange = new EventEmitter();
+  timeRangeValue: TimeRange = 43200;
```

Comment:
I think it is easier to understand if you use `private _timeRange: TimeRange = 43200;` instead of `timeRangeValue: TimeRange = 43200`
Also, since you have a getter/setter for it is should be private as no one should directly access the value.